upgrade_downgrade/test_dual_control_plane_upgrade_boutique.sh

```diff
@@ -0,0 +1,343 @@
+
+writeMsg "Reset cluster"
+copy_test_files
+resetCluster "${TO_ISTIOCTL}"
+
+# Clone the repository
+REPO_PATH=$(mktemp -d "microservices-demo-XXXXX")
+git clone https://github.com/GoogleCloudPlatform/microservices-demo.git "${REPO_PATH}" --depth=1
+
+# Install Initial version of Istio
+writeMsg "Deploy Istio ${FROM_TAG}"
+${FROM_ISTIOCTL} install -f "${TMP_DIR}/iop-control-plane.yaml" -y --revision "${FROM_REVISION}"
+${FROM_ISTIOCTL} install -f "${TMP_DIR}/iop-gateways.yaml" -y --revision "${FROM_REVISION}" 
+waitForPodsReady "${ISTIO_NAMESPACE}"
+
+# 1. Create namespace and label for automatic injection
+# 2. Deploy online boutique application and Istio configuration
+kubectl create namespace "${TEST_NAMESPACE}"
+kubectl label namespace "${TEST_NAMESPACE}" istio-injection- || echo "istio-injection label removed"
+kubectl label namespace "${TEST_NAMESPACE}" istio.io/rev="${FROM_REVISION}"
+
+# It is not as simple as throwing YAML at it. Problem is, there are dependencies between
+# services and loadgenerator should not start before others as it will fall into CrashLoopBackoff
+# So we have to make sure that if something has fallen into CrashLoopBackoff then keep restarting
+# the deployment so that the timeout wouldn't be too long.
+function deploy_boutique_shop_app() {
+  local test_ns="$1"
+  kubectl apply -f "${REPO_PATH}/release/kubernetes-manifests.yaml" -n "${test_ns}"
+  while true; do
+    local pod_count=0
+    local run_count=0
+    for p in $(kubectl get pods -n "${test_ns}" -o name); do
+      pod_count=$((pod_count+1))
+      local pod_line
+      pod_line=$(kubectl get "$p" -n "${test_ns}")
+      if [[ "${pod_line}" == *"Running"* ]]; then
+        run_count=$((run_count+1))
+        continue
+      fi
+      if [[ "$pod_line" == *CrashLoopBackOff* || "$pod_line" == *Error* ]]; then
+        # NOTE: Boutique shop app deployment YAML has a specific pattern
+        # where name of the deployment matches the value of app label. That
+        # is app=<xyz> would have its deployment name as <xyz>
+        local deployment_name
+        deployment_name=$(kubectl get "$p" -n "${test_ns}" -o jsonpath='{.metadata.labels}' | jq -r 'app')
+        echo "$p is stuck in CrashLoopBackoff or Error. So restart deployment ${deployment_name}"
+        kubectl rollout restart deployment "${deployment_name}" -n "${test_ns}"
+      fi
+    done
+    if (( run_count == pod_count )); then
+      echo "Boutique shop deployed successfully"
+      break
+    fi
+    sleep 10
+  done
+}
+
+deploy_boutique_shop_app "${TEST_NAMESPACE}"
+kubectl apply -f "${REPO_PATH}/release/istio-manifests.yaml" -n "${TEST_NAMESPACE}"
+
+# But load-generator should not have a sidecar. So patch its deployment
+# and restart deployment
+kubectl patch deployment loadgenerator -n "${TEST_NAMESPACE}" --patch '{"spec":{"template":{"metadata":{"annotations": {"sidecar.istio.io/inject": "false"}}}}}'
```

Comment:
Yeah. Problem is that I'm doing `kubectl apply -f` . `loadgenerator` is also defined in the same file. I thought of using `--selector` as mentioned [here](
https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply). However, there is no label for `Deployment` resource. So that does not work either :cry: . Even if we manage to put this in a different namespace, pod template should be patched to use `frontend.test:80` instead of `frontend:80`

Reply:
i don't get your point. you can either continue to use -f with no namespace and put the namespace explicitly in the resource in the file, or split it up into multiple files and use -n.

Reply:
Much of this was due to pulling those YAMLs directly. As discussed, I'll copy it from their repo and fix them.